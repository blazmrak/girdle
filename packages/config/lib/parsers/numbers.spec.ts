import { initializeEnvironment } from '../index'
import { expectTypeOf } from 'expect-type'
import { extractError } from '../../tests/helpers/assertions'
import { Var } from './index'

describe('numbers', () => {
  it('parses basic options', () => {
    const env = { NUM: '2', OD: '1' }

    const config = initializeEnvironment(
      {
        defaultNum: Var.number().default(3),
        od: Var.number().default(3),
        optionalNum: Var.number().optional(),
        num: Var.number(),
        namedNum: Var.number().name('NUM'),
      },
      { env },
    )

    expect(config).toMatchSnapshot()
    expectTypeOf(config).toEqualTypeOf<{
      defaultNum: number
      od: number
      optionalNum: number | undefined
      num: number
      namedNum: number
    }>()
  })
  it('Fails on invalid number', () => {
    const env = { NUM: 'abc' }

    const e = extractError(() =>
      initializeEnvironment(
        {
          num: Var.number(),
        },
        { env },
      ),
    )

    expect(e).toMatchSnapshot()
  })
})
