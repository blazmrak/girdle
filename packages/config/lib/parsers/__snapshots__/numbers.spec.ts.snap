// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`number parser Fails on invalid number 1`] = `
EnvironmentInitializationFailed {
  "errors": [
    {
      "error": "Value of 'num' (NUM) should be a number",
    },
  ],
}
`;

exports[`number parser parses basic options 1`] = `
{
  "defaultNum": 3,
  "env": "local",
  "namedNum": 2,
  "num": 2,
  "od": 1,
  "optionalNum": undefined,
}
`;
